aten/src/ATen/native/nested/NestedTensorMath.cpp

```diff
@@ -164,6 +164,32 @@ Tensor NestedTensor_gelu(const Tensor& self, c10::string_view approximate) {
       });
 }
 
+Tensor NestedTensor_nested_tensor_from_mask(const Tensor& t, const Tensor& mask, double min_padding_rate) {
+    TORCH_CHECK(mask.scalar_type() == at::ScalarType::Bool, "Mask tensor should be Bool");
+    TORCH_CHECK(mask.dim() == 2, "Padding mask should be 2D");
+    TORCH_CHECK(t.dim() == 3, "Input should be a 3D tensor, N * L * D");
+    // One assumption is mask is always padded at the end , never in the middle
+    auto N = t.size(0), L = t.size(1), D = t.size(2);
+    auto NN = mask.size(0), LL = mask.size(1);
+    TORCH_CHECK(N == NN && L == LL, "Mask size should match input size");
+    // N * L
+    Tensor sizes = mask;
+    sizes.logical_not_();
```

Comment:
this indicates to me that it's probably better to have `mask=True` be defined to mean "locations to include" rather than "locations to exclude"

Reply:
Yes, but in MHA and BetterTransformer, the current biggest customer of nested tensor, we are using such mask=False by defined, so let's keep it this way.

Reply:
For MHA / BetterTransformer, the mask can be flipped there before passing it to this op, no? That would save an unnecessary flip for other users and improve performance.

`mask=True` for including also fits better with masking semantics used throughout the rest of PyTorch (e.g. [`masked_fill_`](https://pytorch.org/docs/stable/generated/torch.Tensor.masked_fill_.html), ['masked_scatter_'](https://pytorch.org/docs/stable/generated/torch.Tensor.masked_scatter_.html), etc.). It's unfortunate that MHA is an exception to this, but there have been complaints about it and ideally I think we shouldn't follow in its footsteps for a new, generic op.

Reply:
What are those other users?  This is a private function, and was not intended as a new generic op.  I can see the argument of aligning with with other mask definitions, but the innermost loop elements are MHA and Layer, so that's where the most conversions would happen.  

Call it padding, not mask, and let conversion take either mask or padding - with padding being inverse polarity of mask?

Reply:
Arguably what we have here is a description (mask) of padding, not the usual mask of what's "valid". At least call the inputs in the code prominantly padding and before inversion, and mask after inversion if we only support 1 polarity of the mask?